x-pack/docs/en/security/authentication/jwt-realm.asciidoc

```diff
@@ -0,0 +1,335 @@
+as an OIDC _Relying Party_ (RP). In particular the `oidc` realm is the only supported way to enable OpenID
+Connect authentication in Kibana.
+
+TIP: If the front-end application does not exist, use `access` tokens instead. See <<token-authentication-services>>.
+     Other bearer token options are available if accessing {es} directly, without a front-end application.
+     JWT realm is for when that front-end application already exists, and you want to integrate that
+     front-end application with {es}.
+
+A simple JWT realm configuration is as follows:
+
+[source, yaml]
+------------------------------------------------------------
+xpack.security.authc.realms.jwt.jwt1:
+  order: 3 <1>
+  client_authentication.type: shared_secret <2>
+  allowed_issuer: "https://issuer.example.com/jwt/" <3>
+  allowed_audiences: [ "https://app.example.net/" ]
+  allowed_signature_algorithms: [RS256] <4>
+  pkc_jwkset_path: example.jwkset
+  claims.principal: sub <5>
+------------------------------------------------------------
+<1> The realm `order` controls the order in which the configured realms are checked
+    when authenticating a user.
+<2> The `client_authentication.type` setting is described below in <<jwt-client-authentication>>
+<3> The `allowed_issuer` and `allowed_audiences` settings are described below in
+    <<jwt-validation>>
+<4> The `allowed_signature_algorithms` and `pkc_jwkset_path` settings are
+    described below in <<jws-signature-validation>>
+<5> The `claims.principal` setting is described below in <<jwt-authorization>>
+[[external-workflow-requirements]]
+==== External Workflow Requirements
+
+JWT realm functionality is derived from OIDC ID Connect (OIDC) workflows. A high-level understanding of OIDC Tokens helps with JWT realm.
+
+In OIDC workflows, end-users initiates contact with an OIDC Relying Party (RP), and are redirected to authenticate to an OIDC Provider (OP).
+Separate tokens are issued at different steps to represent authentication, authorization, and identification.
+
+.OIDC Tokens
+* Grant Token: End-users authenticate to the OIDC OP, receive an OIDC Grant Token, and present it to the OIDC RP as proof of authentication.
+* ID Token: OIDC RPs ask the OIDC OP to verify the OIDC Grant Token, and exchange it for two tokens. One is an OIDC ID Token.
+* Access Token: OIDC RPs ask the OIDC OP to verify the OIDC Grant Token, and exchange it for two tokens. One is an OIDC Access Token.
+
+IMPORTANT: The OIDC ID Token is a JWT. The encoding is transparent to the OIDC RP, the Grant and Access tokens are opaque values.
+
+When the OIDC RP performs actions on behalf of the end-user, it can make calls into some back-end applications.
+
+.If the back-end application is:
+* part of OIDC OP workflow, the OIDC RP presents the OIDC Access token, and the back-end application asks the OIDC OP to verify authorization.
+* {es} (not part of OIDC OP work), the OIDC RP must present the OIDC ID Token.
+
```

Comment:
I wanted to clarify ID Tokens being transparent, and Grant and Access Tokens being opaque, because a question came up during design about using Access token. If not stated in the docs, I think the question about trying to use Access token instead of ID Token will come up again.

Reply:
I removed the paragraph about OpenID making non-ES calls, and I tried to simplify the remaining test a little bit.